portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public final class BlogHashMap {
+  private static final String[] internalTags = {"#general", "#wellbeing", "#music", "#games"};
+
+  // Takes and puts a list of BlogMessage type in map.
+  // Each BlogMessage type should contain tag that matches tags in the internal System. 
+  public static void putInMap(List<BlogMessage> messages, Map<String, LinkedList<BlogMessage>> map) {
+    for (BlogMessage message : messages) {
+      boolean isTagSupported = false;
+      for (int i = 0; i < internalTags.length; i++){
+        if (message.getTag().equals(internalTags[i])){
+          isTagSupported = true;
+          if (map.containsKey(message.getTag())){
+            map.get(message.getTag()).addLast(message);
+          } else {
+            // No such mapping with specified tag,
+            // Create first mapping with specified tag.
+            LinkedList<BlogMessage> firstValue = new LinkedList<BlogMessage>();
+            firstValue.addFirst(message); 
+            map.put(message.getTag(), firstValue);
+          }
+        } 
+      }
+      if (!isTagSupported) {
+        System.out.println("Tag not supported. Tag: " + message.getTag());
```

Comment:
Prefer "System.err.println" if this is intended to be an error message vs. an info message.

Reply:
Done!